### ISSUE TITLE
Logging to progress bar doesn't work when calling trainer.test()

### ISSUE THREAD

**Issue by minhduc0711** [PyTorchLightning/pytorch-lightning#3518]
## 🐛 Bug

The metric logged by `EvalResult.log()` doesn't show up in the progress bar when setting `prog_bar=True`.

#### Code sample
```
import time
import torch
import torch.nn as nn
import torch.nn.functional as F
from torch.utils.data import TensorDataset, DataLoader
import pytorch_lightning as pl

class MyModel(pl.LightningModule):
    def __init__(self):
        super(MyModel, self).__init__()
        self.layer = nn.Linear(64, 1)

    def forward(self, x):
        return self.layer(x)

    def test_step(self, batch, batch_idx):
        X, y = batch
        loss = F.mse_loss(self.forward(X), y)

        result = pl.EvalResult()
        result.log("test_loss", loss, prog_bar=True)
        time.sleep(1)
        return result

test_ds = TensorDataset(torch.randn(64 * 5, 64), torch.randn(64 * 5, 1))
test_loader = DataLoader(test_ds, batch_size=64)

model = MyModel()
trainer = pl.Trainer(logger=False)
trainer.test(model, test_dataloaders=test_loader)
```

### Expected behavior

`test_loss` shows up in the progress bar when running the above script.

### Environment

 - PyTorch Version: 1.6
 - OS: Linux
 - How you installed PyTorch: pip
 - Python version: 3.8.5

**Comment by awaelchli** [comment]
logging metrics to the progress bar when running over the test or validation set doesn't make sense. You should compute the metric over the whole dataset.

**Comment by gerardsn** [comment]
@awaelchli I'm not sure I understand your answer. If the `prob_bar` flag shouldn't be used during validation or testing, then why does it exist in `EvalResult.log()`?

**Comment by minhduc0711** [comment]
I wanted to quickly check if the batch testing results were sensible or not, so I think that's one possible use case.

But if this is by design, then why even include `on_step` and `on_epoch` in `EvalResult().log`? What are the intended behaviors for these parameters?

**Comment by awaelchli** [comment]
@williamFalcon

**Comment by dragondx** [comment]
Any ways to customize this behavior?
